activejob/lib/active_job/arguments.rb

```diff
@@ -73,9 +73,9 @@ def serialize_argument(argument)
         when Array
           argument.map { |arg| serialize_argument(arg) }
         when ActiveSupport::HashWithIndifferentAccess
-          result = serialize_hash(argument)
-          result[WITH_INDIFFERENT_ACCESS_KEY] = serialize_argument(true)
-          result
+          serialize_indifferent_hash(argument)
+        when defined?(ActionController) && ActionController::Parameters
```

Comment:
I think we can just check if the object respond to `permitted?` This will avoid coupling with Action Controller and make easier to test.

```
when -> (arg) { arg.respond_to?(:permitted) }
  serialize_indifferent_hash(argument.to_h)
```

Probably as the last clause in this `case` since we don't want to run this proc on most of the arguments.